features/step_definitions/addresses-generation-steps.js

```diff
@@ -32,28 +34,29 @@ Then(/^I should see my latest address "([^"]*)" at the top$/, async function (ad
 });
 
 Then(/^I see every generated address is unique$/, async function () {
-  let addressesStringArray = [];
+  const addressesStringArray = [];
   const addresses = await this.driver.findElements(By.xpath("//div[@class='WalletReceive_addressId']"));
 
-  for(let i=1; i<=addresses.length; i++){
-    let addr = await this.driver.findElement(By.css(`.generatedAddress-${i} .WalletReceive_addressId`)).getText().then(function(addr){
-      addressesStringArray.push(addr);
-    });
+  for (let i = 1; i <= addresses.length; i++) {
```

Comment:
the array as a const really got to me, but it’s correct. Maybe this is less ugly?

```
const addressesStringArray = Array.from({ length: addresses.length }, (_, k) => k + 1)
  .map(i => {
    await this.driver
      .findElement(By.css(`.generatedAddress-${i} .WalletReceive_addressId`))
      .getText()
  })
```